src/main/scala/io/opentargets/etl/backend/stringProtein/StringProtein.scala

```diff
@@ -0,0 +1,201 @@
+ |    |    |    |-- interaction_identifier: string (nullable = true)
+ |    |    |    |-- pubmed_id: string (nullable = true)
+ |    |-- interaction_score: long (nullable = true)
+ |-- interactorA: struct (nullable = true)
+ |    |-- biological_role: string (nullable = true)
+ |    |-- id: string (nullable = true)
+ |    |-- id_source: string (nullable = true)
+ |    |-- organism: struct (nullable = true)
+ |    |    |-- mnemonic: string (nullable = true)
+ |    |    |-- scientific_name: string (nullable = true)
+ |    |    |-- taxon_id: long (nullable = true)
+ |-- interactorB: struct (nullable = true)
+ |    |-- biological_role: string (nullable = true)
+ |    |-- id: string (nullable = true)
+ |    |-- id_source: string (nullable = true)
+ |    |-- organism: struct (nullable = true)
+ |    |    |-- mnemonic: string (nullable = true)
+ |    |    |-- scientific_name: string (nullable = true)
+ |    |    |-- taxon_id: long (nullable = true)
+ |-- source_info: struct (nullable = true)
+ |    |-- database_version: string (nullable = true)
+ |    |-- source_database: string (nullable = true)
+
+
+ */
+object StringProtein extends Serializable with LazyLogging {
+
+  def apply(stringDataset: DataFrame, scorethreshold: Int)(implicit ss: SparkSession): DataFrame = {
+    import ss.implicits._
+
+    logger.info("Compute string protein dataset threshold: " + scorethreshold.toString)
+    stringDataset
+      .withColumn("interaction_score", ltrim(col("combined_score")).cast(IntegerType))
+      .filter(col("interaction_score") >= scorethreshold)
+      .filter(col("protein1").contains("9606."))
+      .filter(col("protein2").contains("9606."))
+      .withColumn("id_source_p1", regexp_replace(col("protein1"), "9606.", ""))
+      .withColumn("id_source_p2", regexp_replace(col("protein2"), "9606.", ""))
+      .withColumn("biological_role", lit("unspecified role"))
+      .withColumn("id_source", lit("ensembl_protein"))
+      .withColumn("organism",
+                  struct(lit("human") as "mnemonic",
+                         lit("Homo sapiens") as "scientific_name",
+                         lit("9606").cast("bigint") as "taxon_id"))
+      .withColumn("interactorA",
+                  struct(col("id_source"),
+                         col("biological_role"),
+                         col("id_source_p1") as "id",
+                         col("organism")))
+      .withColumn("interactorB",
+                  struct(col("id_source"),
+                         col("biological_role"),
+                         col("id_source_p2") as "id",
+                         col("organism")))
+      .withColumn("source_info",
+                  struct(lit("11") as "database_version", lit("string") as "source_database"))
+      .withColumn("causal_interaction", lit("False").cast(BooleanType))
+      .drop("protein1", "protein2", "id_source_p1", "id_source_p2", "biological_role", "id_source")
+      .withColumn(
```

Comment:
All these `withColumn` calls looks to me like they are doing the same thing and only updating the name and identifier. Would something like this work?

```
val cols = Seq(("coexpression", "MI:2231"), ("cooccurence", ""MI:2231"), ..., ("database", ""), ("textmining", "MI:0110")
cols.foldLeft(df)((acc, nxt) => { 
  acc.withColumn("e_"+nxt._1, 
       struct(
          lit(nxt._1) as "interaction_detection_method_short_name",
          lit(nxt._2) as "interaction_detection_method_mi_identifier",
          col(nxt._1).cast(LongType) as "evidence_score",
          lit(null) as "interaction_identifier",
          lit(null) as "pubmed_id"
        )
}

I **haven't** tested the above, just an idea to avoid having to repeat the struct so many times.